tracing/opencensus.go

```diff
@@ -81,16 +82,26 @@ func (oct *OpenCensusTracer) ApplyConfig(cfg *config.Config) error {
 	return nil
 }
 
+// Deprecated: Use Shutdown.
 func (oct *OpenCensusTracer) Finish() error {
+	return oct.Shutdown(context.Background())
+}
+
+func (oct *OpenCensusTracer) Shutdown(ctx context.Context) error {
 	err := oct.acquireGlobal()
 	defer octMutex.Unlock()
 	if err != nil {
 		return errors.New("finish called on OpenTracer which is not the global OpenCensusTracer")
 	}
 
 	for _, configOpt := range oct.configOptions {
-		if err = configOpt(nil); err != nil {
-			return err
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
```

Comment:
Is there any reason why we skip performing the shutdown when the context is cancelled?

Reply:
It's a common way to allow for interrupting a long running functions through the context. OpenTelemetry uses this approach as well for interrupting the Shutdown function.